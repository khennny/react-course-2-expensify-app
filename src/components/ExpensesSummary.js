import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => {
    return (
        props.expenses.length > 0 && 
        <p>Viewing {props.expenses.length} expenses totalling {numeral(expensesTotal(props.expenses)/100).format('$0,0.00')}</p>
    )
}

const mapStateToProp = (state) => {    
    return{
        expenses: selectExpenses(state.expenses,{...state.filters})
    }
}

export default connect(mapStateToProp)(ExpenseSummary)
 