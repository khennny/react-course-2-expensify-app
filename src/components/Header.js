import React from 'react';
import {NavLink, Link} from 'react-router-dom';

const Header = () => (
    <header>
        <Link to="/"><h1>Expensify</h1></Link> 
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>                       
    </header>
);

export default Header;