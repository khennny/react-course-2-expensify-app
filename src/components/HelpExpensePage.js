import React from 'react';

const HelpExpensePage = () =>(
    <div>
        This is from my Help Expense Component
    </div>
);

export default HelpExpensePage;