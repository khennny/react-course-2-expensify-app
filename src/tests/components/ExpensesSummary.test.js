import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpensesSummary';
import expenses from '../fixture/expenses';

test('should render ExpensesSummary with 1 expense with total',() => {
    const Newexpenses = [expenses[0]];
    const wrapper = shallow(<ExpenseSummary expenses={Newexpenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 2 expenses with total',() => {
    const Newexpenses = expenses;
    const wrapper = shallow(<ExpenseSummary expenses={Newexpenses}/>);
    expect(wrapper).toMatchSnapshot();
});