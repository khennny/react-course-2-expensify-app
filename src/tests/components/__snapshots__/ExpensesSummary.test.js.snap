// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense with total 1`] = `
<p>
  Viewing 
  1
   expenses totalling 
  $1.95
</p>
`;

exports[`should render ExpensesSummary with 2 expenses with total 1`] = `
<p>
  Viewing 
  3
   expenses totalling 
  $1,141.95
</p>
`;
